### ISSUE TITLE
chart export breaks when showPointLabels is true

### ISSUE THREAD

**Issue by teefal** [primefaces/primefaces#1525]
After days of debugging, I've narrowed down a tricky problem using Primefaces 5.3/6.0 charts.  My app has several charts that get exported as images for use in a PDF.  All was working until I added one line that setShowPointLabels( true ).  The export method then never returned, eventually timing out.  Oddly enough, the problem didn't occur using Internet Explorer, but it did in Chrome on both Mac and Windows.

Using the showcase samples, I've narrowed things down ... 

```
package com.example;

import org.primefaces.model.chart.*;

import javax.annotation.PostConstruct;
import javax.faces.bean.ManagedBean;
import java.io.Serializable;

@ManagedBean
public class BarChartExport implements Serializable {

    private BarChartModel barModel;

    @PostConstruct
    public void init() {
        createBarModels();
    }

    public BarChartModel getBarModel() {
        return barModel;
    }

    private void createBarModels() {
        barModel = initBarModel();
        barModel.setTitle("Bar Chart");
        barModel.setLegendPosition( "ne" );
        barModel.setShowPointLabels( true );  // <-- CHANGE to false and it works
        Axis xAxis = barModel.getAxis(AxisType.X);
        xAxis.setLabel( "Gender" );
        Axis yAxis = barModel.getAxis(AxisType.Y);
        yAxis.setLabel("Births");
        yAxis.setMin( 0 );
        yAxis.setMax( 200 );
    }

    private BarChartModel initBarModel() {
        BarChartModel model = new BarChartModel();

        ChartSeries boys = new ChartSeries();
        boys.setLabel("Boys");
        boys.set("2004", 120);
        boys.set("2005", 100);
        boys.set("2006", 44);
        boys.set("2007", 150);
        boys.set("2008", 25);

        ChartSeries girls = new ChartSeries();
        girls.setLabel("Girls");
        girls.set("2004", 52);
        girls.set("2005", 60);
        girls.set("2006", 110);
        girls.set("2007", 135);
        girls.set("2008", 120);

        model.addSeries(boys);
        model.addSeries(girls);

        return model;
    }
}

```

```
<!DOCTYPE html PUBLIC '-//W3C//DTD XHTML 1.0 Transitional//EN'
        'http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd'>

<html xmlns='http://www.w3.org/1999/xhtml'
      xmlns:h='http://java.sun.com/jsf/html'
      xmlns:ui='http://java.sun.com/jsf/facelets'
      xmlns:p='http://primefaces.org/ui'
      xmlns:f='http://java.sun.com/jsf/core'>

<h:head>
</h:head>

<h:body>
    <p:chart type="bar" model="#{barChartExport.barModel}" style="width:500px;height:300px"
            widgetVar="chart"/>

    <p:commandButton type="button" value="Export" icon="ui-icon-extlink"
                     onclick="exportChart()"/>
    <p:dialog widgetVar="dlg" showEffect="fade" modal="true" header="Chart as an Image">
        <p:outputPanel id="output" layout="block" style="width:500px;height:300px"/>
    </p:dialog>

    <script type="text/javascript">

        function exportChart() {
            //export image
        $('#output').empty().append(PF('chart').exportAsImage());

            //show the dialog
        PF('dlg').show();
        }
    </script>

</h:body>
</html>
```

**Comment by teefal** [comment]
Simplified things by removing the jqplot testing stuff.

**Comment by mertsincan** [comment]
Can you please try this issue with 5.3.x or 6.0.x version?

**Comment by teefal** [comment]
How do I do this without being pro/elite?  It says I need a password.  Build from git?

**Comment by mertsincan** [comment]
> Build from git?

Yes, you can build primefaces project in our github. Then, you can use produced jar for testing your issue.

**Comment by teefal** [comment]
Confirmed the bug happens with newly built version of the  trunk.  Attached find my stripped down sample project, which is run under a vanilla Glassfish 3.1.2.2 domain, freshly created with no params.

[primebug.zip](https://github.com/primefaces/primefaces/files/381516/primebug.zip)

Not sure how to check 5.3.x ... not familiar with github enough to grab the old source.

**Comment by teefal** [comment]
Btw, the bug happens on Mac/Chrome and Win10/Chrome, but not Win10/IE11, Win10/Edge or either Mac/Win10 Safari or Firefox.

**Comment by mertsincan** [comment]
Thanks for the sample project and details. Can you please attach a screenshot for us? I tested with Win8.1/Chrome. My screenshot;

![chrome_test](https://cloud.githubusercontent.com/assets/11868120/17104018/a20f1714-5289-11e6-9205-6878be86713a.PNG)

**Comment by teefal** [comment]
My screen looks exactly like yours, except there's no box.  It hangs before the box opens, eventually timing out.

**Comment by teefal** [comment]
This is the area where it hangs, by the way ... it's about to call _jpqToImage on div.jqplot-point.label.jqplot-series-0.jqplot-point-11

![jpqtoimage](https://cloud.githubusercontent.com/assets/2871552/17106448/58f0682e-5259-11e6-828e-68334afac6ae.png)

**Comment by teefal** [comment]
More specifically, it's the call to writeWrappedText inside that invocation that hangs.

**Comment by teefal** [comment]
FYI, my version of Mac/Chrome is Version 46.0.2490.71 (64-bit).

**Comment by rainermueller** [comment]
Issue still exists using PrimeFaces 6.1.5 and Chrome Version 61.0.3163.100 running on OSX 10.12.6.

**Comment by melloware** [comment]
Have we determined if this is a straight jQPlot issue by taking PrimeFaces out of the equation altogether? I know there have been other issues reported around exporting...

https://github.com/jqPlot/jqPlot/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aopen%20export

**Comment by sworcester-hsdmetrics** [comment]
We are experiencing the same issue in our application. Any solutions / workarounds from PrimeFaces on this issue?

**Comment by melloware** [comment]
@teefal @sworcester-hsdmetrics @rainermueller  Attached is my test you can run using "mvn clean jetty:run" using the latest PF 6.3-SNAPSHOT.  Everything is working fine in Chrome, IE11, Edge, and Firefox with this sample.
[pf-1525.zip](https://github.com/primefaces/primefaces/files/2676132/pf-1525.zip)

So maybe the JQPlot update has fixed this issue. Can you verify and close?

**Comment by melloware** [comment]
@tandraschko @Rapster Safe to mark this ticket as fixed for 6.3 and close it.  I have created a test case that proves its fixed.
